CMakeLists.txt

```diff
@@ -83,6 +90,47 @@ else()
     message(FATAL_ERROR "Unknown OS type")
 endif()
 
+if(NOT UMF_LINK_HWLOC_STATICALLY)
+    pkg_check_modules(LIBHWLOC hwloc>=2.3.0)
+    if(NOT LIBHWLOC_FOUND)
+        find_package(LIBHWLOC 2.3.0 REQUIRED hwloc)
+    endif()
+    # add PATH to DLL on Windows
+    set(DLL_PATH_LIST
+        "${DLL_PATH_LIST};PATH=path_list_append:${LIBHWLOC_LIBRARY_DIRS}/../bin"
+    )
+else()
+    if(NOT WIN32)
+        message(FATAL_ERROR "hwloc can be static linked only on Windows")
+    endif()
```

Comment:
as this only works on Release and we don't have an easy way to check this on Windows, you could add a message, similarly to proxy_lib (`"The proxy library will be built, however it is supported only in the Release build on Windows"`).

I'm not sure what are the consequences in Debug build - it won't work...?

Reply:
It won't build due to many workarounds to force it to build during configure+build time.
The only way is to add an extra step for building dependencies using ExternalProject_Add(), then the whole change will be limited to one function and Debug/Release will work like a charm. If you prefer such a solution, please, talk to Rafał and I can make a separate PR the same day.